runtime/compiler/env/j9method.cpp

```diff
@@ -8392,9 +8399,10 @@ TR_J9ByteCodeIlGenerator::runFEMacro(TR::SymbolReference *symRef)
                TR::Node *receiverHandle = pop();
                if (receiverHandle->getOpCode().hasSymbolReference() && receiverHandle->getSymbolReference()->getSymbol()->isArrayShadowSymbol())
                   {
-                  if (thunkDetails->isShareable())
+                  if (!knotEnabled || thunkDetails->isShareable())
```

Comment:
I have a general comment about this—not something that needs to be addressed in the code here. If we're somehow able to generate custom thunks without a known object table, and if we're actually doing so, we should stop. Custom thunks are pointless without known objects.

Reply:
Sorry, I don't know enough about custom thunks to tell if what you described is happening.